openpdf/src/main/java/com/lowagie/text/pdf/PdfWriter.java

```diff
@@ -487,22 +484,12 @@ void writeCrossReferenceTable(OutputStream os, PdfIndirectReference root, PdfInd
                 PdfStream xr = new PdfStream(buf.toByteArray());
                 buf = null;
                 xr.flateCompress(writer.getCompressionLevel());
-                xr.put(PdfName.SIZE, new PdfNumber(size()));
-                xr.put(PdfName.ROOT, root);
-                if (info != null) {
-                    xr.put(PdfName.INFO, info);
-                }
-                if (encryption != null)
-                    xr.put(PdfName.ENCRYPT, encryption);
-                if (fileID != null)
-                    xr.put(PdfName.ID, fileID);
+                xr.putAll(trailer);
```

Comment:
Why do you make this change?
Pkease comment on why you made all these changes.

Reply:
Yep, I agree. It seems support for 2 GB files (that is `int -> long` conversion) is intermixed with other changes. That makes it harder to review it...

Reply:
In the existing code:
* If object compression is disabled, `writeCrossReferenceTable` writes an `xref` table without the following `trailer`. The caller (PdfWriter or PdfStamperImp) then calls `PdfTrailer.toPdf`, which writes the `trailer` and finally the `startxref`.
* If object compression is enabled, `writeCrossReferenceTable` writes an `/XRef` stream which incorporates all the same properties that a `trailer` would. Since the trailer has effectively already been written, PdfWriter does not call `PdfTrailer.toPdf`, but instead just writes the `startxref` itself.

This relates to writing large files because objects with offsets > 9,999,999,999 require an `/XRef` stream, rather than an `xref` table - so `writeCrossReferenceTable` now needs to write an `/XRef` stream if either object compression is enabled *or* if the file is sufficiently large.

I did not want to duplicate this logic in multiple places. So I simplified the code by moving `trailer` writing into `writeCrossReferenceTable`. This makes sense anyway - an old-style `xref` is inseparable from its following `trailer`; you can't write one without the other. So now `writeCrossReferenceTable` writes either `xref`+`trailer` or the equivalent `/XRef` stream, and the caller doesn't need to worry about which one it wrote.

The first commit in this PR is just the trailer refactoring, and then the second commit does the `int -> long` conversion, so you can review them one at a time.